### PR TITLE
goto-analyzer: tweak user interface

### PR DESCRIPTION
**Pull request by kroening**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: c62ca10e65d31f1ac95aeac40b664561ae19160a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88639406
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 86938c17dab3272c0cf7df5c1cd88df5ec0ede75).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88639623

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 8dbac5d0643173819df2f7cc1b56ff5bf096ebad).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88640384
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 3f58dfbc909d0c4edf2e2505baa2a1ca0356259f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88640621

File: src/goto-analyzer/goto_analyzer_parse_options.cpp [hunk 1]
```diff
@@ -229,8 +229,7 @@ void goto_analyzer_parse_optionst::get_command_line_options(optionst &options)
   }
   else if(cmdline.isset("intervals") || cmdline.isset("non-null"))
   {
-    // For backwards compatibility either of these on their own means show
```
Comment (martin-cs) [inline src/goto-analyzer/goto_analyzer_parse_options.cpp]: The key here is this part.  ```--intervals``` and ```--non-null``` on their own are ```--show``` for backwards compatibility.  If we can break that then the command handling can be significantly simplified.

**Review by martin-cs — CHANGES_REQUESTED** [review]
If we're going to break backwards compatibility of options and change every test case then let's do it properly and wipe out all of the default and backwards compatible options.

**Comment by kroening** [comment]
@martin-cs Agreed, but this will need to be a separate PR; the current one already exceeds the usual size limit.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 9ad9b057a5bd87479663852c2bdac0c9b09237f0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89437831

**Comment by martin-cs** [comment]
Why do we need to change the default behaviour of these again?

**Comment by kroening** [comment]
The current behaviour is good for developers of abstract domains, but not useful for "regular users". We should prioritise the users. The developers can be asked to use a command-line option.

**Comment by martin-cs** [comment]
Then why just these two domains?  Why don't we just change the default task and make this all consistent?  If we are making a 'usability' and 'least surprise' argument then why does this only apply to these two domains?

**Comment by kroening** [comment]
@martin-cs Agreed, should be for all -- now done!

**Review by allredj — COMMENTED** [review]
:warning:
Diffblue compatibility check is currently unavailable.
Please create manual bump.
(cbmc commit: 44888273eb99485409b4bd9935efe55cfb7feb45).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90729665

**Review by allredj — COMMENTED** [review]
:warning:
Diffblue compatibility check is currently unavailable.
Please create manual bump.
(cbmc commit: 52e7cb3fe2a5a98f8b6dddee9491ebb451d9ac43).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90733299

**Review by chrisr-diffblue — COMMENTED** [review]
I'm fine with the change to make --verify the default behaviour - that makes sense. I'm a bit less convinced by the change to the output that drops the filename from the messages. Dropping the function name I'm relaxed about, but dropping the file name seems like it might be making life less consistent for end users? especially if scripting around the tools?

**Comment by peterschrammel** [comment]
Scripting around tools should use json or XML.

**Comment by peterschrammel** [comment]
Nonetheless, I don't understand the rationale behind dropping the filename. This doesn't seem user-friendly.

**Comment by kroening** [comment]
The file name is still there: the properties are grouped by function, and for each function, the file name is printed.

**Comment by chrisr-diffblue** [comment]
Thanks for the clarification - I'd missed that, possibly because I'm not seeing a test that demonstrates the filename being there? I was basing my (flawed, as it turned out) reasoning on the fact that the diffs for the various updated regressions tests just showed the filename being removed, with no sign of it being anywhere else in the regex matches.

**Review by martin-cs — COMMENTED** [review]
The changes to the console output -- fine, can merge the last two commits.
The idea of verify as default -- also fine.
The current implementation of verify as default -- breaks some other things.

I'll put together a PR for how I think it should be done.  We should merge the other two commits though.

**Comment by martin-cs** [comment]
https://github.com/diffblue/cbmc/pull/3437 is how I think the default should be changed.

**Comment by kroening** [comment]
Ok, will get #3437 merged, and will then rebase.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 353b96b579620136ff5620990917551d00947ad9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91972608

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 74617cf01fc3b9c7981b6a3be2af6b7e96b4f75a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/94083795

File: regression/goto-analyzer/intervals_10/test.desc [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
---intervals --verify
+--intervals
 ^EXIT=0$
 ^SIGNAL=0$
-^\[main.assertion.1\] file main.c line 8 function main, assertion j<=100: Success$
-^\[main.assertion.2\] file main.c line 11 function main, assertion j<101: Success$
-^\[main.assertion.3\] file main.c line 14 function main, assertion j>100: FAILURE \(if reachable\)$
-^\[main.assertion.4\] file main.c line 17 function main, assertion j<99: UNKNOWN$
-^\[main.assertion.5\] file main.c line 20 function main, assertion j==100: FAILURE \(if reachable\)$
+^\[main.assertion.1\] line 8 assertion j<=100: Success$
```
Comment (peterschrammel) [inline regression/goto-analyzer/intervals_10/test.desc]: SUCCESS ? (twice)

Comment (kroening) [reply regression/goto-analyzer/intervals_10/test.desc]: done

**Review by peterschrammel — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1278a83a95ad73b539dadcf5852101ffe46e4eb3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/94352768

**Review by martin-cs — APPROVED** [review]
I don't think the changes to the tests are strictly needed as we should be getting people towards the task / analyser / domain / output usage model but beyond that, looks good.

**Pull request merged**